contracts/contracts/SimpleReadAccessController/library.cairo

```diff
@@ -0,0 +1,57 @@
+%lang starknet
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.starknet.common.syscalls import get_tx_info
+from starkware.cairo.common.bool import TRUE, FALSE
+
+from SimpleWriteAccessController.library import simple_write_access_controller
+
+namespace simple_read_access_controller:
+    func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        owner_address : felt
+    ):
+        simple_write_access_controller.constructor(owner_address)
+        return ()
+    end
+
+    func has_access{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        user : felt, data_len : felt, data : felt*
+    ) -> (bool : felt):
+        let (has_access) = simple_write_access_controller.has_access(user, data_len, data)
+        if has_access == TRUE:
+            return (TRUE)
+        end
+
+        # NOTICE: access is granted to account contracts, to enable off-chain reads.
+        # The account abstraction architecture can be used to deploy a custom contract
+        # that could access the data and pass it forward to a contract,
+        # but a workflow like that would be of little use
+        # because the contract would need to trust user inputs as it can't verify the data itself.
+        # While the financial contract could accept calls from verified account contracts only (to verify data is correct),
+        # it would be a risky strategy to depend on this part of infrastructure that can easily be upgraded.
+        let (tx_info) = get_tx_info()
+        if tx_info.account_contract_address == user:
+            return (TRUE)
+        end
+
+        return (FALSE)
+    end
+
+    # TODO: remove when starkware adds get_class_hash_at
```

Comment:
How would we use `get_class_hash_at` here?

Reply:
The we won't need to have a copy of `SimpleWriteAccessController::check_access` here in `SimpleReadAccessController`. In `SimpleWriteAccessController::check_access` we could fetch class_hash and use [library_call](https://www.cairo-lang.org/docs/hello_starknet/calling_contracts.html#library-calls) which would execute has_access in corresponding context: if class_hash return hash of `SimpleReadAccessController` (or parent) then we will execute `SimpleReadAccessController::has_access` otherwise we will execute `SimpleWriteAccessController::has_access`

Reply:
(last part & c++ example of 3 here)[https://github.com/smartcontractkit/chainlink-starknet/pull/49]

Reply:
Shall we leave or remove this TODO ?

Reply:
Cairo 0.9.0 added this so we should fix the TODO: start using `get_class_hash_at`

Reply:
That's an API it's not present in syscalls as [get_caller_address](https://github.com/starkware-libs/cairo-lang/blob/13cef109cd811474de114925ee61fd5ac84a25eb/src/starkware/starknet/common/syscalls.cairo#L196) for example

Reply:
So if we want to change it we would have to change interface adding extra argument for `class_hash`

Reply:
Let's keep the TODO then